h/views.py

```diff
@@ -113,6 +118,68 @@ def stream(context, request):
         return context
 
 
+@view_config(layout='app', route_name='atom_stream',
+             renderer='h:templates/stream.atom')
+def atom_stream(context, request):
```

Comment:
I'd be inclined to move all of the communication with the API into its own package (i.e. out of the view function).

Similarly, I'd be inclined to move the piece that turns an API response into a template context suitable for generating a feed into its own module.

In the end I'd hope that the view function ends up looking something like:

```python
def atom_stream(request):
    try:
        resp = api_client.get("/search", ...)
    except Timeout:
        request.response.status_code = 504
        return ...

    rows = resp.json()["rows"]
    return feed.from_search_results(rows)
```

This way the pieces underneath become largely pure, and can be tested without having to play with Pyramid view machinery.